additional_user.tf

```diff
@@ -19,11 +19,11 @@ data "template_file" "additional_user" {
   template = <<EOF
 info "Creating user:"
 # The printf string is put in single-quotes because it may contain it's own double-quotes.
-printf '  Login: \"$${user_login}\"\n  Gecos: \"$${user_gecos}\"\n  Shell: \"$${user_shell}\"\n  Supplemental Groups: \"$${user_supplemental_groups}\"\n  Authorized Keys: \"$${user_authorized_keys}\"\n'
+printf '  Login: \"$${user_login}\"\n'
 
 # The ssh_keys variable is put in single-quotes because it may contain it's own double-quotes.
-if [ "$${user_login}x" == "x" -o '$${user_authorized_keys}x' == "x" ] ; then
```

Comment:
Thanks for avoiding my mistake / unintended additional expansion of the SSH key here!

Reply:
I actually had to change this back to something close to what you already had. So, here's the unfortunate thing: you can't reference bash variables inside a template_file resource in Terraform, because the bash variable always gets interpolated by terraform. So even though my check written like `if [ -z $variable}` looks correct, that variable will get interpolated and filled by terraform no matter what. I tried to even `export` the variable to another variable name and then use that, but then terraform still picks it up and complains that I haven't stated a value for the variable. This is a really unfortunate shortcoming of templating bash inside of terraform. 

So,  I updated it to use the comparison check with the `x` inserted, like you had already.